content/docs/rfcs/36/README.md

```diff
@@ -106,6 +106,105 @@ Fields:
 - `data`: Decrypted message payload base64 encoded,
 - `padding`: Padding base64 encoded.
 
+### `FilterSubscription` type
+
+The criteria to create subscription to a light node in JSON Format:
+
+```ts
+{
+    contentFilters: ContentFilter[];
+    topic: string?;
+}
+```
+
+Fields:
+
+- `contentFilters`: Array of [`ContentFilter`](#contentfilter-type) being subscribed to / unsubscribed from.
+- `topic`: Optional message topic.
+
+
+### `ContentFilter` type
+
+```ts
+{
+    contentTopic: string;
+}
+```
+
+Fields:
+
+- `contentTopic`: The content topic of a waku message.
+
+### `StoreQuery` type
+
+Criteria used to retreive historical messages
+
+```ts
+interface StoreQuery {
+    pubsubTopic?: string;
+    contentFilters?: ContentFilter[];
+    startTime?: number;
+    endTime?: number;
```

Comment:
```suggestion
   timeFilter?: {
       startTime: number;
       endTime: number;
   }
```

You have to set both or neither. To avoid user error I recommend modifying the API so that the user cannot make a mistake without breaking typing.

Reply:
According to this code, it should be possible to set either:
https://github.com/status-im/nwaku/commit/2cb7123df84b18ab7e4df0be0d77353cca4820f2#diff-9adb9e1f6697143d9e2f5e5870293666b221fe21047b4ca05e426130329068c7R223-R231

Reply:
> According to this code, it should be possible to set either: [status-im/nwaku@2cb7123#diff-9adb9e1f6697143d9e2f5e5870293666b221fe21047b4ca05e426130329068c7R223-R231](https://github.com/status-im/nwaku/commit/2cb7123df84b18ab7e4df0be0d77353cca4820f2#diff-9adb9e1f6697143d9e2f5e5870293666b221fe21047b4ca05e426130329068c7R223-R231)

Well, this code needs to be fixed :)

From https://rfc.vac.dev/spec/21/
> A time-based query is considered valid if its end_time is larger than or equal to the start_time. Queries that do not adhere to this condition will not get through **e.g. an open-end time query in which the start_time is given but no end_time is supplied is not valid.** If both start_time and end_time are omitted then no time-window filter takes place.*

Cc @LNSD